app/models/host_facet_base.rb

```diff
@@ -0,0 +1,43 @@
+class HostFacetBase < ActiveRecord::Base
+  self.abstract_class = true
+
+  belongs_to_host
+
+  # Add facet's details to host's ENC by returning it from this method
+  # Basic ENC hash structure:
+  # ---
+  # classes:
+  #   ...
+  # parameters:
+  #   ...
+  # environment:
+  #   ...
+  def info
+    {}
+  end
+
+  # Specify any smart proxy id's used by this facet.
+  def smart_proxy_ids
+    []
+  end
+
+  # Add any filters to template selection returned from host
+  def template_filter_options(kind)
+    {}
+  end
+
+  # Add search criteria for finding a configuration template.
+  def provisioning_template_options
+    {}
+  end
+
+  # Change attributes that will be sent to an facet based on inherited values from the hostgroup.
+  def self.inherited_attributes(hostgroup, facet_attributes)
```

Comment:
`facet_attributes` is the part of facet's attributes from `apply_inherited_attributes`. The implementation looks like this:
``` Ruby
def apply_inherited_attributes(attributes, initialized = true)
#...
  Facets.registered_facets.values.each do |facet_config|
    facet_attributes = attributes["#{facet_config.name}_attributes"]
    facet_attributes = facet_config.model.inherited_attributes(new_hostgroup, facet_attributes)
    attributes["#{facet_config.name}_attributes"] = facet_attributes if facet_attributes
  end
#...
```